src/server/routes/revision.js

```diff
@@ -237,14 +238,25 @@ router.get('/:id', (req, res, next) => {
 					formatWorkChange
 				)
 			);
-			const props = {revision: revision.toJSON(), diffs};
-			res.render('page', {
+			const props = Object.assign({}, req.app.locals, res.locals, {
+				revision: revision.toJSON(),
 				title: 'RevisionPage',
-				markup: ReactDOMServer.renderToString(RevisionPage(props))
+				diffs
 			});
+			const markup = ReactDOMServer.renderToString(
+				<Layout {...props}>
```

Comment:
There's some duplication here of what the controller is doing. Is there any way around that? It'd be nice if the controller could take the appropriate props and either render to string or into the DOM, but I'm not sure if that's feasible. @Arthelon, @LordSputnik, feedback appreciated.